analysis/quantifier-comparisons/11-five-prime-benchmarking.Rmd

```diff
@@ -0,0 +1,372 @@
+---
+title: "Alevin-Fry Five Prime"
+author: "Ally Hawkins for CCDL"
+output: 
+  html_notebook:
+    toc: true
+    toc_float: true
+---
+
+This notebook compares using Alevin-fry to pre process samples captured using the 5' kit from 10X rather than the usual 3' approach. 
+I am comparing Alevin-fry, using the decided upon parameters we have chosen for analyzing the 3' libraries, to Cell Ranger for 2 5' samples. 
+
+Alevin-fry is run using selective alignment to the splici index using the `cr-like-em` resolution for UMI deduplication. 
+Additionally, to signify that we are aligning a 5' library, the read orientation was switched from forward to reverse during the `generate-permit-list` step. 
+
+In contrast, Cell Ranger is able to automatically detect the library type based on how many of the reads are in the antisense direction. 
+
+Here, we are testing how well Alevin-fry using the `rc` option with 5' library performs in comparison to using Cell Ranger's automatic detection. 
+To do this we will look at the distribution of UMI/cell, genes/cell, correlation of mean gene expression, and number of detected genes that overlap, similar to previous benchmarking comparisons. 
+
+## Set Up
+ 
+
+```{r}
+library(magrittr)
+library(ggplot2)
+library(SingleCellExperiment)
+library(ggupset)
+```
+
+
+```{r}
+# path to results files with sces and qc dataframes from running benchrmarking_generate_qc_df.R
+base_dir <- here::here()
+file_dir <- file.path(base_dir, "data", "five-prime-rc", "results")
+
+# qc files 
+quant_info_file <- file.path(file_dir, "quant_info.tsv")
+coldata_df_file <- file.path(file_dir, "coldata_qc.tsv")
+rowdata_df_file <- file.path(file_dir, "rowdata_qc.tsv")
+
+# sce objects
+fry_file <- file.path(file_dir, "splici_salign_cr-like-em_sces.rds")
+cellranger_file <- file.path(file_dir, "cellranger_sces.rds")
+```
+
+```{r}
+# read in dataframes needed for plotting
+quant_info <- readr::read_tsv(quant_info_file)
+coldata_df <- readr::read_tsv(coldata_df_file)
+rowdata_df <- readr::read_tsv(rowdata_df_file)
+```
+
+```{r}
+# read in sces
+fry_sce <- readr::read_rds(fry_file)
+cellranger_sce <- readr::read_rds(cellranger_file)
+```
+
+```{r}
+# merge coldata df with quant_info
+coldata_info_df <- coldata_df %>%
+  # rename tool to be either cellranger or alevin-fry
+  dplyr::mutate(tool = ifelse(tool == "splici_salign_cr-like-em", "alevin-fry", "cellranger")) %>%
+  dplyr::left_join(quant_info,
+                   by = c("tool" = "tool", 
+                          "quant_id" = "quant_dir"))
+```
+
+## Per Cell Comparisons
+
+### All Cells
+
+In order to see if there are any differences between Alevin-fry and Cell Ranger in the ability to assign reads to each cell barcode, we first are looking at all cells before restricting to shared cells only. 
+
+```{r}
+# mito content in shared cells
```

Comment:
Is it worth commenting about that the mito percent is so much higher for alevin-fry? My first thought was that this might be a pre-filtering artifact, but it seems to hold post-filtereing too.

Reply:
One quick thought here — not sure if it's worth digging into or not.  CellRanger is mapping against their custom reference, and so it has the entire mt chromosome (including genes on there that may not be part of the standard annotation).  In this case, reads will map to the best location, which may be an unannotated (or at least not part of the restricted annotation) gene on the mt chromosome.  In that case, it won't get counted toward an annotated mt gene and won't show up in the QC.

On the other hand, alevin-fry knowns only about the mt transcripts that are explicitly included in the reference.  I know for a fact that there are mito transcripts missing from the standard annotations (like MT-RNR2 and MT-RNR1).  Possibly then, reads coming from similar mt transcripts could be mapped instead to some annotated mt transcript and counted as mt expression.  In this case, the STAR/CellRanger approach would actually have similar mito expression, but it just wouldn't show up in the count matrix since the mt chromosome is essentially acting as a "decoy" for these reads (because the transcripts of origin aren't annotated in the count matrix).  It might be worth looking in the CellRanger BAM and what kind of mapping rate you get to the mt chromosome, and does that signify more reads arising from there than are accounted for in the reported gene counts?  This may not be what is going on of course, but this *is* a thought I've had on this a few times before.

Reply:
Thanks for the clear explanation @rob-p. I think you bring up an interesting point and potentially something to look into. Mitochondrial genes also tend to have a lot of homology as well correct? I wonder if this is something that is more commonly seen in using the `cr-like-em` resolution rather than just the `cr-like` resolution and that some of the increased detection is due to multimapped reads not being thrown away here.